docs/architecture-and-design.md

```diff
@@ -171,12 +171,12 @@ Examples of the Visitor Pattern can be found with the following classes:
 
 
 ## Observer Design Pattern
-The Observer Design Pattern is useful for letting components of a system "notify" each other when something happens. This saves us having to waste CPU time polling components for updates.
+The Observer Design Pattern is useful for letting components of a system "notify" each other when something happens. Read [https://www.geeksforgeeks.org/observer-pattern-set-1-introduction/] for a general introduction to the pattern.
 
-Read [https://www.geeksforgeeks.org/observer-pattern-set-1-introduction/] for more information.
-
-We use the Observer Design Pattern to connect all our top-level modules in the system as shown in the [Architecture Overview](#architecture-overview). The components can notify each other and provide data whenever it is available. For example, the [AI](#ai) will "observer" the [Backend](#backend) so that when the backend gets new information, it can notify the [AI](#ai) and the [AI](#ai) can choose to take action with the new data.
+Our implementation of this pattern consists of two classes, `Observer` and `Subject`. `Observer`'s can be registered with a `Subject`, after which new values will be sent from each `Subject` to all of it's registered `Observer`'s. Please see the headers of both classes for details. Note that a class can extend both `Observer` and `Subject`, thus receiving and sending out data. In this way we can "chain" multiple classes.
```

Comment:
`Observer`'s -> `Observer`s

Reply:
Changed and gave a quick example. @jonathanlew - just a note that the example will need to be updated when the "Sensor Fusion" layer exists.